### DISCUSSION TITLE
Remove need to copy custom serializers to every assembly

### DISCUSSION THREAD

**Discussion by mbstudio2004** [FirstGearGames/FishNet#594] [Feature Request]
So im a big fan of assembly definitions and i organize my project in multiple assemblies, but as those assemblies sometimes rely on each other, for example one assembly is interaction system other assembly is player, and when doing server side interaction i need to send which player did it, so i write a custom network serializer.

Problem relies in fact that i need to write all of serializers in every assembly that will use , so in interaction assembly i need to write custom serializer for player, and in player for player. Its cluttery, but for now it works

I have  a potential temporary solution, so, that you have one sepparate assembly where its sole purpose is to put serializers there if you use custom serializers, or idk, maybe a folder in project that is in main assembly so it can reference any custom and only putting there serializers. 

Maybe settings for code generation, because some people dont use assemblies, so it works as now
maybe option just to take serializers from a folder, or from one asssembly
to be able to choose.  Until a better solution comes.

**Comment by FirstGearGames** [comment]
Not possible as this is a Unity limitation.
You do not need to copy the full serializer, you can create one that redirects to your full one.

**Reply by mbstudio2004** [reply]
Could you just elaborate the redirection process. How exactly im redirecting? You mean, writing a proxy script that just runs main script??

**Reply by FirstGearGames** [reply]
Basically.
Pseudo code...

```cs
//Full serializer....
public static void WriteMyType(this Writer w, MyType value)
{
//things to do full write.
}

//Redirect writer in another assembly.
public static void WriteMyTypeRedirect(this Writer w, MyType value) => WriteMyType(w, value);
```
This would of course mean everything would need to see the assembly with the full writer, which is in a way what you suggested. Unfortunately though this is all we have with a Unity limitation blocking our desired behavior of global custom serializers.

**Reply by ooonush** [reply]
I remember we had a discussion about this on discord a couple weeks ago. At that time we came to the conclusion that this limitation can be bypassed by using Source Generator. No news on that?

**Comment by FirstGearGames** [comment]
@ooonush We found a way and this was actually added recently-ish as a feature. 
In 4.1.3+ 
```
Added ability for custom serializers to work across assemblies.
Added UseGlobalCustomSerializer attribute. Use on classes which you made a custom serializer for to make serializer work across assemblies.
```
